thaw/src/input/mod.rs

```diff
@@ -53,20 +53,22 @@ pub fn Input(
     #[prop(optional)] comp_ref: ComponentRef<InputRef>,
     #[prop(optional, into)] class: OptionalProp<MaybeSignal<String>>,
     #[prop(attrs)] attrs: Vec<(&'static str, Attribute)>,
+    #[prop(optional, into)] parser: OptionalProp<Callback<String, String>>,
+    #[prop(optional, into)] formatter: OptionalProp<Callback<String, String>>,
 ) -> impl IntoView {
     let theme = use_theme(Theme::light);
     mount_style("input", include_str!("./input.css"));
 
     let value_trigger = create_trigger();
     let on_input = move |ev| {
         let input_value = event_target_value(&ev);
+        parser.map_or_else(|| value.set(input_value.clone()), |c| value.set(c.call(input_value.clone())));
```

Comment:
`value.set` should be placed after the allow_value logic, because allow_value is to prevent disallowed values ​​in advance.

Reply:
Yes, you a right. Only when parser is present the input_value should be parsed before the allow_value logic because input_value may contain disallowed chars. Do you agree?

Reply:
Agreed. Could be changed to the following:

```rust
let mut input_value = event_target_value(&ev);
if let Some(parser) = parser {
     input_value = parser.call(input_value);
}

/// allow_value logic ...

value.set(input_value);
```